tests/store/tracking/test_file_store.py

```diff
@@ -533,12 +534,15 @@ def test_create_experiment(self):
 
     def test_create_experiment_appends_to_artifact_uri_path_correctly(self):
         cases = [
-            ("path/to/local/folder", "path/to/local/folder/{e}"),
+            ("path/to/local/folder", "{cwd}/path/to/local/folder/{e}"),
             ("/path/to/local/folder", "/path/to/local/folder/{e}"),
-            ("#path/to/local/folder?", "#path/to/local/folder?/{e}"),
-            ("file:path/to/local/folder", "file:path/to/local/folder/{e}"),
+            ("#path/to/local/folder?", "{cwd}/#path/to/local/folder?/{e}"),
+            ("file:path/to/local/folder", "{cwd}/file:path/to/local/folder/{e}"),
             ("file:///path/to/local/folder", "file:///path/to/local/folder/{e}"),
-            ("file:path/to/local/folder?param=value", "file:path/to/local/folder/{e}?param=value"),
+            (
+                "file:path/to/local/folder?param=value",
+                "{cwd}/file:path/to/local/folder?param=value/{e}",
```

Comment:
@dbczumar ..and here. Notice, how the query param is now resolved as a literal path too compared to the existing test. To be specific
this relative path is now resolved as `file:path/to/local/folder?param=value` is resolved as `{cwd}/file:path/to/local/folder?param=value/{e}`
versus existing assert of this `file:path/to/local/folder?param=value` to `file:path/to/local/folder/{e}?param=value`

Reply:
if I don't parse that a literal and attempt a URI parsing for it, I can technically grab fragment and query params and pass those along to the absolute artifact URI.

Reply:
This should be `file:{cwd}/path/to/local/folder?param=value/{e}` because the supplied artifact location is a file URI with a relative path. We need to resolve the path to an absolute path and substitute it in the URIs path component, instead of prepending the current working directory to the URI.